tests/phpunit/tests/admin/wpMediaListTable.php

```diff
@@ -109,6 +109,51 @@ public function tear_down() {
 		parent::tear_down();
 	}
 
+	/**
+	 * Tests that column_author() displays the correct author link when an author exists.
+	 *
+	 * @ticket 12345
+	 * @covers WP_Media_List_Table::column_author
+	 */
+	public function test_column_author_with_valid_author() {
+		wp_set_current_user( self::$admin );
+
+		$author_id = self::factory()->user->create( array( 'role' => 'author' ) );
+		$post      = self::factory()->post->create_and_get( array( 'post_author' => $author_id ) );
+
+		$author_name  = get_the_author_meta( 'display_name', $author_id );
+		$expected_url = esc_url( add_query_arg( array( 'author' => $author_id ), 'upload.php' ) );
+
+		ob_start();
+		self::$list_table->column_author( $post );
+		$output = ob_get_clean();
+
+		$this->assertStringContainsString( '<a href="' . $expected_url . '">', $output );
+		$this->assertStringContainsString( '>' . esc_html( $author_name ) . '</a>', $output );
+	}
+
+	/**
+	 * Tests that column_author() displays the correct fallback when no author exists.
+	 *
+	 * @ticket 12345
+	 * @covers WP_Media_List_Table::column_author
+	 */
+	public function test_column_author_with_no_author() {
+		wp_set_current_user( self::$admin );
+
+		add_filter( 'the_author', '__return_empty_string' );
```

Comment:
If you add a unit test for this, I think it should create a new attachment post without an author (possibly setting the author ID as `0`?). I suggested using the filter to reproduce the empty link, but replacing the author with an empty string is not a case that I would expect anyone to encounter in a real site.